src/lessons/Data-Structures/04.md

```diff
@@ -0,0 +1,51 @@
+
+So far we've exclusively been discussing adorable images for the sake of our general
+merriment, but content addressing can be used on all different types of files and data,
+from JSON objects to term papers to videos. For cryptographic hashing to work, we need
+to know what data format we're working with and use an appropriate tool.
+
+## Decoding data structures
+
+A `CID` is a particular form of content addressing. It's a specification developed for
+IPFS, a specific decentralized web protocol that we'll discuss in later tutorials,
+but it has very broad implications.
+
+A `CID` is a single identifier that contains a cryptographic hash **and**
+information about how to interpret that data called a "codec." (Codecs encode,
+decode, compress, and decompress data in certain formats. You use them every time
```

Comment:
I'd leave compress and decompress out for now. Technically, there's not compression in the current codecs, and we're trying to do compression at another layer (transport and storage). This will be quite confusing for anyone trying to map these semantics on to what they already know about HTTP so I'd just leave compression out for now.